extensions/panache/hibernate-reactive-panache-kotlin/runtime/src/main/kotlin/io/quarkus/hibernate/reactive/panache/kotlin/PanacheCompanion.kt

```diff
@@ -0,0 +1,805 @@
+     *
+     * @return the number of entities deleted.
+     * @see .delete
+     * @see .delete
+     * @see .delete
+     */
+    @CheckReturnValue
+    @GenerateBridge
+    fun deleteAll(): Uni<Long> {
+        throw INSTANCE.implementationInjectionMissing()
+    }
+
+    /**
+     * Delete an entity of this type by ID.
+     *
+     * @param id the ID of the entity to delete.
+     * @return false if the entity was not deleted (not found).
+     */
+    @CheckReturnValue
+    @GenerateBridge
+    fun deleteById(id: Any): Uni<Boolean> {
+        throw INSTANCE.implementationInjectionMissing()
+    }
+
+    /**
+     * Delete all entities of this type matching the given query, with optional indexed parameters.
+     *
+     * WARNING: the default implementation of this method uses a bulk delete query and ignores
+     * cascading rules from the JPA model.
+     *
+     * @param query a [query string][io.quarkus.hibernate.reactive.panache]
+     * @param params optional sequence of indexed parameters
+     * @return the number of entities deleted.
+     * @see .deleteAll
+     * @see .delete
+     * @see .delete
+     */
+    @CheckReturnValue
+    @GenerateBridge
+    fun delete(query: String, vararg params: Any): Uni<Long> {
+        throw INSTANCE.implementationInjectionMissing()
+    }
+
+    /**
+     * Delete all entities of this type matching the given query, with named parameters.
+     *
+     * WARNING: the default implementation of this method uses a bulk delete query and ignores
+     * cascading rules from the JPA model.
+     *
+     * @param query a [query string][io.quarkus.hibernate.reactive.panache]
+     * @param params [Map] of named parameters
+     * @return the number of entities deleted.
+     * @see .deleteAll
+     * @see .delete
+     * @see .delete
+     */
+    @CheckReturnValue
+    @GenerateBridge
+    fun delete(query: String, params: Map<String, Any>): Uni<Long> {
+        throw INSTANCE.implementationInjectionMissing()
+    }
+
+    /**
+     * Delete all entities of this type matching the given query, with named parameters.
+     *
+     * WARNING: the default implementation of this method uses a bulk delete query and ignores
+     * cascading rules from the JPA model.
+     *
+     * @param query a [query string][io.quarkus.hibernate.reactive.panache]
+     * @param params [Parameters] of named parameters
+     * @return the number of entities deleted.
+     * @see .deleteAll
+     * @see .delete
+     * @see .delete
+     */
+    @CheckReturnValue
+    @GenerateBridge
+    fun delete(query: String, params: Parameters): Uni<Long> {
+        throw INSTANCE.implementationInjectionMissing()
+    }
+
+    /**
+     * Persist all given entities.
+     *
+     * @param entities the entities to persist
+     * @return
+     * @see .persist
+     * @see .persist
+     * @see .persist
+     */
+    @CheckReturnValue
+    @GenerateBridge(callSuperMethod = true)
+    fun persist(entities: Iterable<*>): Uni<Void> {
```

Comment:
this method is bound to `Entity` on the other impls.  same for the overloads as well.